tardis/montecarlo/packet_source.py

```diff
@@ -38,45 +50,107 @@ def create_packet_seeds(self, no_of_packets, seed_offset):
+    def set_state_from_model(self, model):
         pass
 
-    def create_zero_limb_darkening_packet_mus(self, no_of_packets):
-        """
-        Create zero-limb-darkening packet :math:`\mu` distributed
-        according to :math:`\\mu=\\sqrt{z}, z \isin [0, 1]`
+    @abc.abstractmethod
+    def create_packet_radii(self, no_of_packets, *args, **kwargs):
+        pass
+
+    @abc.abstractmethod
+    def create_packet_nus(self, no_of_packets, *args, **kwargs):
+        pass
+
+    @abc.abstractmethod
+    def create_packet_mus(self, no_of_packets, *args, **kwargs):
+        pass
+
+    @abc.abstractmethod
+    def create_packet_energies(self, no_of_packets, *args, **kwargs):
+        pass
+
+    def create_packets(self, no_of_packets, *args, **kwargs):
+        """Generate packet properties as arrays
 
         Parameters
         ----------
         no_of_packets : int
-            number of packets to be created
+            Number of packets
+
+        Returns
+        -------
+        array
+            Packet radii
+        array
+            Packet frequencies
+        array
+            Packet directions
+        array
+            Packet energies
         """
+        radii = self.create_packet_radii(no_of_packets, *args, **kwargs)
+        nus = self.create_packet_nus(no_of_packets, *args, **kwargs)
+        mus = self.create_packet_mus(no_of_packets, *args, **kwargs)
+        energies = self.create_packet_energies(no_of_packets, *args, **kwargs)
 
-        # For testing purposes
-        if montecarlo_configuration.LEGACY_MODE_ENABLED:
-            return np.sqrt(np.random.random(no_of_packets))
-        else:
-            return np.sqrt(self.rng.random(no_of_packets))
+        return radii, nus, mus, energies
+
+
+class BlackBodySimpleSource(BasePacketSource):
+    """
+    Simple packet source that generates Blackbody packets for the Montecarlo
+    part.
 
-    def create_uniform_packet_energies(self, no_of_packets):
+    Parameters
+    ----------
+    radius : float64
+        Initial packet radius
+    temperature : float
+        Absolute Temperature.
+    base_seed : int
+        Base Seed for random number generator
+    legacy_secondary_seed : int
+        Secondary seed for global numpy rng (Deprecated: Legacy reasons only)
+    """
+
+    @classmethod
+    def from_model(cls, model, *args, **kwargs):
+        return cls(model.r_inner[0], model.t_inner.value, *args, **kwargs)
+
+    def __init__(self, radius=None, temperature=None, **kwargs):
+        self.radius = radius
+        self.temperature = temperature
+        super().__init__(**kwargs)
+
+    def set_state_from_model(self, model):
         """
-        Uniformly distribute energy in arbitrary units where the ensemble of
-        packets has energy of 1.
+        Set state of packet source (correct state should be ensured before creating packets)
+        """
+        self.radius = model.r_inner[0]
+        self.temperature = model.t_inner.value
```

Comment:
See my comment on the notebook regarding astropy quantities. However, if this is consistent with how it was previously  AND the r_inner is stored without quantities, that is okay (but tells us that we need to look into this in the future).

Reply:
Yeah it's consistent with what it was before, but yes we need to change everything to quantities later.